File: SDKs/amazonaws.com/rds/2013-09-09/go/pkg/models/shared/dbsecuritygroups.go
package shared

type DbSecurityGroups struct {
	DBSecurityGroupDescription *string
	DBSecurityGroupName        *string
	Ec2SecurityGroups          []Ec2SecurityGroupList
	IPRanges                   []IPRangeList
	OwnerID                    *string
	VpcID                      *string
}

File: SDKs/amazonaws.com/rds/2014-09-01/typescript/src/sdk/models/shared/dbsecuritygroups.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Ec2SecurityGroupList } from "./ec2securitygrouplist";
import { IpRangeList } from "./iprangelist";



export class DbSecurityGroups extends SpeakeasyBase {
  @SpeakeasyMetadata()
  dbSecurityGroupDescription?: string;

  @SpeakeasyMetadata()
  dbSecurityGroupName?: string;

  @SpeakeasyMetadata({ elemType: Ec2SecurityGroupList })
  ec2SecurityGroups?: Ec2SecurityGroupList[];

  @SpeakeasyMetadata({ elemType: IpRangeList })
  ipRanges?: IpRangeList[];

  @SpeakeasyMetadata()
  ownerId?: string;

  @SpeakeasyMetadata()
  vpcId?: string;
}
